client/reducers/scheduleReducers.js

```diff
@@ -0,0 +1,68 @@
+import { createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+  studentId: null,
+  studentName: '',
+  teacherData: [
+    {
+      name: '',
+      availability: [
+        '6:00PM - 6:10pm',
+        '6:10PM - 6:20pm',
+        '6:20PM - 6:30pm',
+        '6:30PM - 6:40pm',
+        '6:40PM - 6:50pm',
+        '6:50PM - 7:00pm',
+      ],
+    },
+  ],
+  appointmentTime: 'None',
+};
+
+export const scheduleSlice = new createSlice({
+  name: 'schedule',
+  initialState,
+  reducers: {
+    handleChange: (state, action) => {
+      state.appointmentTime = action.payload;
+    },
+
+    submit: (state, action) => {
+      const { teacherName, appointmentTime, studentName } = action.payload;
+      for (let i = 0; i < state.teacherData.length; i++) {
+        if (state.teacherData[i].name === teacherName) {
+          // SA Do we want to keep this if statement to make sure the appointment is still there incase another user selected the same time?
```

Comment:
It would be a cool stretch feature to implement to hold the time in the database as soon as it is selected by user